lms/djangoapps/certificates/apis/v0/views.py

```diff
@@ -187,8 +191,10 @@ class CertificatesListView(GenericAPIView):
                 "download_url": "http://www.example.com/cert.pdf",
                 "grade": "0.98"
             }]
-    """
-
+        """,
+))
+class CertificatesListView(GenericAPIView):
```

Comment:
👍  to @kdmccormick 's [suggestion for an inner class instead of method decorators here](https://github.com/edx/edx-platform/pull/21207#issuecomment-523921534), that looks much nicer.  Is this doable?  Is it possible to pull the default description from the class docstring?

Reply:
Docstrings can be used to generate the "description" for an endpoint, but will it fully populate the swagger.yaml file with parameters, etc?

I'm not sure how to answer this, honestly.  In some ways, this comment amounts to, "Can you do it all completely differently?"  I'm not averse to using docstrings, but we need to see it fleshed out.

Reply:
@nedbat I guess what I, and possibly @pomegranited, are asking is not "Can you do it all completely differently?" but instead "Can we develop some clever tools to avoid devs having to work with Swagger's (in my opinion) unpleasant decorator syntax, while still using Swagger underneath?"

The good thing about this request is it doesn't mean we're asking for blocking changes to this PR. Such convenience decorators/functions/whatever could be gradually added to `openapi_tools` library as the desire/need for them arises, an effort to which I'd happily contribute.

Reply:
And, I wasn't thinking that the parameters and other metadata would be somehow extracted in the docstring. In my original comment, I was imagining that they are defined as class-level attributes in an inner class, which could be processed by a decorator to so that Swagger understands them.